docs/architecture-decisions/multiple-flag-set-support.md

```diff
@@ -0,0 +1,162 @@
+{
+  "$schema": "https://flagd.dev/schema/v1/flagsets.json",
+  "flagSets": {
+    "my-project-1": {
+      "metadata": {
+        ...
+      },
+      "flags": {
+        "my-flag-1": {
+          "metadata": {
+            ...
+          },
+          ...
+        },
+        ...
+      },
+      "$evaluators": {
+        ...
+      }
+    },
+    "my-project-2": {
+      ...
+    }
+  }
+}
+````
+
+We propose to introduce a 3rd json schema `flagSets.json`, which references to `flags.json`:
+
+1. flagSets.json (new)
+2. flags.json
+3. targeting.json
+
+We don't want to support merging of flag sets, due to implementation efforts & potential confusing behaviour of the
+merge strategy.
+Therefore, we propose for the initial implementation, `flagSetId`s must be unique across different sources or the configuration is considered invalid.
+In the future, it might be useful to support and implement multiple "strategies" for merging flagSets from different sources, but that's beyond the scope of this proposal.
+
+### New Data Structure
+
+The storage layer in `flagd` requires refactoring to better support multiple flag sets within one source.
+
+````go
+package store
+
+type State struct {
+    FlagSets map[string]FlagSet `json:"flagSets"` // key = flagSetId
+}
+
+type FlagSet struct {
+    Flags    map[string]model.Flag `json:"flags"` // key = flagKey
+    Metadata Metadata              `json:"metadata,omitempty"`
+}
+
+type Flag struct {
+    State          string          `json:"state"`
+    DefaultVariant string          `json:"defaultVariant"`
+    Variants       map[string]any  `json:"variants"`
+    Targeting      json.RawMessage `json:"targeting,omitempty"`
+    Metadata       Metadata        `json:"metadata,omitempty"`
+}
+
+type Metadata = map[string]interface{}
+````
+
+### OpenFeature Provider Implications
+
+Currently, creating a new flagd provider can look like follows:
+
+````java
+final FlagdProvider flagdProvider =
+        new FlagdProvider(FlagdOptions.builder()
+                .resolverType(Config.Evaluator.IN_PROCESS)
+                .host("localhost")
+                .port(8015)
+                .selector("myFlags.json")
+                .build());
+````
+
+* With the proposed solution the `flagSetId` should be passed to the builder as selector argument instead of the source.
+* `null` is now a valid selector value, referencing flags which do not belong to a flag set. The default/fallback `flagSetId` should be `null`.
+
+### Consequences
+
+* Good, because it decouples flag sets from the sources
+* Good, because we will refactor the flagd storage layer (which is currently storing duplicate data & difficult to
+  understand)
+* Good, because we can support backwards compatibility with the v0 schema
+* Good, because the "null" flag set is logically treated as any other flag set, reducing overall implementation complexity.
+* Bad, because there's additional complexity to support this new config schema as well as the current.
```

Comment:
```suggestion
* Bad, because there's additional complexity to support this new config schema as well as the current.
* Bad, because this is a breaking change in the behavior of the `selector` member.
```